src/test/lambda/vue/samInvokeComponent.test.ts

```diff
@@ -0,0 +1,170 @@
+/*!
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as assert from 'assert'
+import * as util from '../../webviews/util'
+import { render, screen, fireEvent, within } from '@testing-library/vue'
+import { anything, verify, when, instance } from 'ts-mockito'
+import { SamInvokeWebview } from '../../../lambda/vue/samInvokeBackend'
+import Component from '../../../lambda/vue/samInvokeComponent.vue'
+import { capture } from '../../utilities/mockito'
+import { getSettingsPanel } from '../../webviews/components/settingsPanel.test'
+import { WebviewClient } from '../../../webviews/client'
+
+// Convenience functions to cast types based off their tag, should be moved to a shared file
+// There may also be a library for this?
+
+function isSelectElement(element: HTMLElement | null): element is HTMLSelectElement {
+    return element?.tagName === 'SELECT'
+}
+
+function isOption(element: HTMLElement | null): element is HTMLOptionElement {
+    return element?.tagName === 'OPTION'
+}
+
+describe('samInvokeComponent', function () {
+    function createCodeDebugConfig() {
+        return {
+            name: 'foo',
+            invokeTarget: {
+                lambdaHandler: '',
+                projectRoot: '',
+                target: 'code' as const,
+            },
+            request: '',
+            type: '',
+        }
+    }
+
+    let client: WebviewClient<SamInvokeWebview['protocol']>
+    const runtimes = ['nodejs14.x', 'go1.x']
+
+    beforeEach(async function () {
+        client = util.mockClient<SamInvokeWebview>()
+
+        // These commands are called in the component's `created` lifecycle event so they should
+        // always be mocked with something
+        when(client.init()).thenResolve(createCodeDebugConfig())
+        when(client.getRuntimes()).thenResolve(runtimes)
+
+        // Render the component and pass in the mock client
+        // Note that we can't leverage any type-safety here since Typescript does not understand '.vue' files
+        render(Component, { props: { client: instance(client) } })
+
+        // Components can load data asychronously, so we add a short delay to give Vue time to re-render
+        await new Promise(r => setTimeout(r, 25))
```

Comment:
Should be able to remove this by finding or adding an event that's emitted on initial load.